### PR TITLE
lshw.py:Failed commands are: ['lshw -disable network|grep -i interface']

### PR DESCRIPTION
**Pull request by shirishaganta1**
disable option is failing as the condition is failing on powernv. 
So changing the if condition in order to work for both PowerVM and PowerNV

### CODE REVIEW

**Comment by shirishaganta1** [comment]
Before:
=========
JOB ID     : 00729bced0f17e49cc0204791e1569356b5a435a
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2023-02-22T09.12-00729bc/job.log
 (01/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw: STARTED
 (01/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw:  PASS (1.60 s)
 (02/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_short: STARTED
 (02/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_short:  PASS (6.86 s)
 (03/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_class: STARTED
 (03/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_class:  PASS (14.42 s)
 (04/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_verification: STARTED
 (04/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_verification:  PASS (2.41 s)
 (05/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_gen_rep: STARTED
 (05/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_gen_rep:  PASS (85.92 s)
 (06/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_businfo: STARTED
 (06/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_businfo:  PASS (1.43 s)
 (07/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_sanitize: STARTED
 (07/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_sanitize:  PASS (23.06 s)
 (08/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_prod_id_serial: STARTED
 (08/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_prod_id_serial:  PASS (2.29 s)
 (09/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_options: STARTED
 (09/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_options:  FAIL: lshw: Failed commands are: ['lshw -disable network|grep -i interface'] (6.21 s)
 (10/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_notime: STARTED
 (10/10) /home/avocado-fvt-wrapper/tests/avocado-misc-tests/ras/lshw.py:Lshwrun.test_lshw_notime:  PASS (1.32 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

After:
======
(01/10) lshw.py:Lshwrun.test_lshw: STARTED
 (01/10) lshw.py:Lshwrun.test_lshw: PASS (1.95 s)
 (02/10) lshw.py:Lshwrun.test_lshw_short: STARTED
 (02/10) lshw.py:Lshwrun.test_lshw_short: PASS (8.84 s)
 (03/10) lshw.py:Lshwrun.test_lshw_class: STARTED
 (03/10) lshw.py:Lshwrun.test_lshw_class: PASS (18.56 s)
 (04/10) lshw.py:Lshwrun.test_lshw_verification: STARTED
 (04/10) lshw.py:Lshwrun.test_lshw_verification: PASS (3.08 s)
 (05/10) lshw.py:Lshwrun.test_gen_rep: STARTED
 (05/10) lshw.py:Lshwrun.test_gen_rep: PASS (61.60 s)
 (06/10) lshw.py:Lshwrun.test_businfo: STARTED
 (06/10) lshw.py:Lshwrun.test_businfo: PASS (1.79 s)
 (07/10) lshw.py:Lshwrun.test_sanitize: STARTED
 (07/10) lshw.py:Lshwrun.test_sanitize: PASS (13.29 s)
 (08/10) lshw.py:Lshwrun.test_prod_id_serial: STARTED
 (08/10) lshw.py:Lshwrun.test_prod_id_serial: PASS (2.91 s)
 (09/10) lshw.py:Lshwrun.test_lshw_options: STARTED
 (09/10) lshw.py:Lshwrun.test_lshw_options: PASS (5.45 s)
 (10/10) lshw.py:Lshwrun.test_lshw_notime: STARTED
 (10/10) lshw.py:Lshwrun.test_lshw_notime: PASS (1.71 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/10867505/job.log)

File: ras/lshw.py [hunk 1]
```diff
@@ -210,14 +210,18 @@ def test_lshw_options(self):
...
+                                                    "interface\""):
             self.is_fail += 1
-            self.fail_cmd.append("lshw -disable network|grep -i interface")
-        if "interface" not in self.run_cmd_out("lshw -enable network| "
-                                               "grep -i interface"):
+            self.fail_cmd.append(
+                "lshw -disable network|grep -i \"ethernet interface\"")
+        if "Ethernet interface" not in self.run_cmd_out("lshw -enable network| "
+                                                        "grep -i \"ethernet " 
+                                                        "interface\""):
```
Comment (abdhaleegit) [inline ras/lshw.py:221]: the command fails on both powevm and powernv

lshw -enable network| ""grep -i \"ethernet " "interface\"
grep:  interface": No such file or directory
$ echo $?
2

Comment (abdhaleegit) [reply ras/lshw.py:221]: a better rule is to not use shell pipe in python code, how about

cmd = "lshw -enable network"
for res in self.run_cmd_out(cmd).split():
    if "Ethernet interface" not in:

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: ras/lshw.py (hunk 1 above)
Comment (shirishaganta1) [reply ras/lshw.py:221]: it is working as expected Abdul
#  lshw -disable network| grep  -i "ethernet interface"
#  lshw -enable network| grep  -i "ethernet interface"
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
             description: Ethernet interface
#

**Comment by shirishaganta1** [comment]
Addressed the requested changes and resent the PR

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Pull request merged**